algorithms/src/sorting/Kokkos_SortPublicAPI.hpp

```diff
@@ -77,130 +30,30 @@ namespace Kokkos {
-  }
-
-  using CompType = BinOp1D<ViewType>;
-  BinSort<ViewType, CompType> bin_sort(
-      view, CompType(max_bins, result.min_val, result.max_val), sort_in_bins);
-  bin_sort.create_permute_vector(exec);
-  bin_sort.sort(exec, view);
-}
-
-#if defined(KOKKOS_ENABLE_ONEDPL)
-template <class DataType, class... Properties>
-void sort(const Experimental::SYCL& space,
+void sort([[maybe_unused]] const ExecutionSpace& exec,
           const Kokkos::View<DataType, Properties...>& view) {
+  // constraints
   using ViewType = Kokkos::View<DataType, Properties...>;
-  static_assert(SpaceAccessibility<Experimental::SYCL,
-                                   typename ViewType::memory_space>::accessible,
-                "SYCL execution space is not able to access the memory space "
-                "of the View argument!");
-
-  // Can't use Experimental::begin/end here since the oneDPL then assumes that
-  // the data is on the host.
-  static_assert(
-      ViewType::rank == 1 &&
-          (std::is_same<typename ViewType::array_layout, LayoutRight>::value ||
-           std::is_same<typename ViewType::array_layout, LayoutLeft>::value),
-      "SYCL sort only supports contiguous rank-1 Views.");
-
-  if (view.extent(0) == 0) {
-    return;
-  }
-
-  auto queue  = space.sycl_queue();
-  auto policy = oneapi::dpl::execution::make_device_policy(queue);
-  const int n = view.extent(0);
-  oneapi::dpl::sort(policy, view.data(), view.data() + n);
-}
-#endif
-
-// clang-format off
-template <class ExecutionSpace, class DataType, class... Properties>
-std::enable_if_t<
-  (Kokkos::is_execution_space<ExecutionSpace>::value) &&
-  (SpaceAccessibility<
-     HostSpace, typename Kokkos::View<DataType, Properties...>::memory_space
-    >::accessible)
-  >
-// clang-format on
-sort(const ExecutionSpace&, const Kokkos::View<DataType, Properties...>& view) {
-  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
   static_assert(ViewType::rank == 1,
                 "Kokkos::sort: currently only supports rank-1 Views.");
+  static_assert(SpaceAccessibility<ExecutionSpace, MemSpace>::accessible,
+                "Kokkos::sort: execution space instance is not able to access "
+                "the memory space of the "
+                "View argument!");
 
-  if (view.extent(0) == 0) {
+  if (view.extent(0) <= 1) {
     return;
   }
-  auto first = Experimental::begin(view);
-  auto last  = Experimental::end(view);
-  std::sort(first, last);
-}
-
-#if defined(KOKKOS_ENABLE_CUDA)
-template <class DataType, class... Properties>
-void sort(const Cuda& space,
-          const Kokkos::View<DataType, Properties...>& view) {
-  using ViewType = Kokkos::View<DataType, Properties...>;
-  static_assert(ViewType::rank == 1,
-                "Kokkos::sort: currently only supports rank-1 Views.");
 
-  if (view.extent(0) == 0) {
-    return;
+  if constexpr (SpaceAccessibility<HostSpace, MemSpace>::accessible) {
```

Comment:
Is this really the condition we want to use? If the `MemSpace` is CUDA UVM, we probably want to run on the device or is it always faster to sort on the host?

Reply:
good point, not really sure - I will fix this to call cuda also for CUDA UVM. Because that is what happened before, i guess my constraint changed this .

Reply:
You also need to check for `CudaHostPinnedSpace`. Instead of the memory space, would it make more sense to check  on the execution space? Right now we check the memory space but we pass the execution space around. Instead, if the execution space is `Serial`, `OpenMP`, or `Threads` we use `std::sort` and otherwise we use `sort_device_view_without_comparator`?

Reply:
https://github.com/kokkos/kokkos/pull/6239/commits/2a1c79bf8daba86594327d6490b59ea7328eb177